src_c/color.c

```diff
@@ -1679,6 +1679,15 @@ _color_set_length(pgColorObject *color, PyObject *args)
 {
     int clength;
 
+    if (PyErr_WarnEx(
+            PyExc_DeprecationWarning,
+            "You can selectively unpack the rgba values you want, without "
+            "having to set length.\nExample: r, g, b, _ = pygame.Color(200, "
+            "100, 42)\nSo, set_length() will be deprecated in pygame 2.1.3",
+            1) == -1) {
+        return NULL;
+    }
```

Comment:
Maybe it would be best to not try to explain why in the warning.

I also dislike the wording of "will be deprecated in pygame 2.1.3" in several places. It's not something that's happening in the future, if someone is seeing it it's already happened.